### ISSUE TITLE
Changes in configuration.js are not reflected on docs/index.html

### ISSUE THREAD

**Issue by hetraeus** [ziahamza/webui-aria2#472]
Changes in configuration.js are not reflected on docs/index.html. I just followed README.md but it's not working, I have to change port and insert the secret token every time and changed settings are not stored. What am I missing?

**Comment by anthonyraymond** [comment]
Yep, the new refactor broke the old configuration.js file. I'm not sure how you should handle this now. May i ask why you are using `configuration.js` file?

**Comment by anthonyraymond** [comment]
If i were you i'd just forget about the configuration.js and i'd fill the authentications on the webui form. These informations are saved into a cookie (or localstorage) anyway.

**Comment by no1xsyzy** [comment]
If my memory is correct, there was a similar issue.
`configuration.js` will not be read every time. It will only be read for the first time (when nothing persisted locally). If you have ever opened WebUI once, the modification to `configuration.js` will not work.

**Comment by foreverhex** [comment]
I have the same question, the previous configuration.js doesn't working any more. I was using a local folder and open html directly from browser.

Also seems like the url mentioned in README doesn't work? 

https://ziahamza.github.io/webui-aria2/docs

Thanks.

**Comment by anthonyraymond** [comment]
@pratikborsadiya Do you see how we could fix that?

**Comment by YanceyChiew** [comment]
Just open the app.js, search " (c = p && p.auth && p.auth.token ? p.auth.token : null),", find it and modify it to “(c = "xxxx-there is your own token-xxxx"),”.

It will work in anywhere, but you can't modify the token through webpage anymore. which meaning nothing for users who opens index.html from file explorer.

在app.js中搜索上面提到的那行代码，修改成上面提到的格式，这样做对于直接从文件打开index.html有效，也没有别的影响，唯一的问题只是在输入token的元素内不会显示占位符，它直接就是空的。

>             if (g && u.length)
>               return (
>                 (g = !1),
>                 (p = u[0]),
>                 // (c = p && p.auth && p.auth.token ? p.auth.token : null),//modify this line
>                 (c = "yourToken"),
>                 e.init(p),
>                 void (h = setTimeout(v, t))
>               );

**Comment by no1xsyzy** [comment]
I think webpack provides something called like "code-split", which enables manually setup splitting point.
This could be used to load some library code from CDN or separate file. Analogically, maybe we can use this function to treat `configuration.js` as a library and keep it unchanged.

**Comment by YanceyChiew** [comment]
[app.js.txt](https://github.com/ziahamza/webui-aria2/files/2459018/app.js.txt)

Well, I uploaded my own modified app.js as a txt file, which once was modified for supporting packing this program into a chrome extension. It's almostly can save cookies in everywhere now, include the file:// protocol.

But remember, in this way your token won't be clear unless you do it yourself.(Open the debug page and clear localstorage).

**Comment by pratikborsadiya** [comment]
@foreverhex @hetraeus will you guys tell me that why are you guys specifically using webui from file. If it's because of being able to use the app offline I would like to inform you that the [latest version of webui](https://ziahamza.github.io/webui-aria2/docs) hosted on GitHub pages is ready to be used offline once you have visited the application while you are online.

**Comment by Neo-Oli** [comment]
@pratikborsadiya For me it's just that I like self hosting my own stuff. I don't trust a remotely hosted version of this, because it could potentially be compromised and used for malicious purposes (like downloading stuff I don't want).

**Comment by lychichem** [comment]
@pratikborsadiya And for me is that github.io is not always avalible in mainland China, what if the cache of PWA is outdated and the github.io site is unavalible? So we need to use locale files, at least we can deal with this problem when it happens.

**Comment by Rivers47** [comment]
I am also having this issue, it is not storing connection settings to local storage (which seems to be another issue?). Clearing cookie does not help neither. Modifying app.js works for me.

**Comment by recolic** [comment]
I add some `fuck` randomly in `src/js/services/configuration.js` but no error reported while runing `node node-server.js`. Is configuration.js ever executed???

**Comment by pratikborsadiya** [comment]
@recolic `configuration.js` will not be executed directly as we are building the source code into one single file.  It will reflect your changes if you compile the source code after making change using the steps mentioned [here](https://github.com/ziahamza/webui-aria2/blob/master/CONTRIBUTING.md#get-started).

**Comment by recolic** [comment]
> @recolic `configuration.js` will not be executed directly as we are building the source code into one single file. It will reflect your changes if you compile the source code after making change using the steps mentioned [here](https://github.com/ziahamza/webui-aria2/blob/master/CONTRIBUTING.md#get-started).

Thanks! It works.

Could you please add these contents into README.md? It can be confusing if the user has no experience on nodejs. By the way, currently I'm using `npm run dev` to run my server. But I don't want to compile it again and again on every run...

**Comment by pratikborsadiya** [comment]
> I'm using `npm run dev` to run my server. But I don't want to compile it again and again on every run...
You can use `npm run build` to compile source files for production build.

**Comment by recolic** [comment]
@pratikborsadiya i see. but how do I run built project?

**Comment by pratikborsadiya** [comment]
You can run `node node-server.js` just like you were running before making the change.
